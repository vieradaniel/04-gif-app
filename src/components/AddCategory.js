import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [InputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        
        setInputValue(e.target.value);
        
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        if(InputValue.trim().length >2 ){
            setCategories (cats => [InputValue,...cats]);
            setInputValue("");
        }
    }

  return (
        
        <form onSubmit={handleSubmit}>
            <p>{InputValue}</p>
            <input 
                type="text" 
                value = { InputValue }
                onChange = { handleInputChange }
            />
        </form>
        
        );
};

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
