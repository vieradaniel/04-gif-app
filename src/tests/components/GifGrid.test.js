import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>',()=>{
    const category= 'dragon ball';

    test('should show the correct component and match with the snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });
 
    test('should change the snapshot when useFetchGifs is executed', () => {
        
        const gifs = [
        {
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        },
        {
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }
    ]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
 
        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
 
    });




})