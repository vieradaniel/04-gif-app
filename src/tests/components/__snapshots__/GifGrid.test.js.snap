// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> should change the snapshot when useFetchGifs is executed 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dragon ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dragon ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
